pkg/plugins/diff/delta_differ.go

```diff
@@ -0,0 +1,96 @@
+package tablediff
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/treeverse/lakefs/pkg/plugins"
+	"github.com/treeverse/lakefs/pkg/plugins/internal"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type DeltaLakeDiffer struct {
+	client TableDifferClient // This is the underlying GRPC API client
+}
+
+func (d *DeltaLakeDiffer) Diff(ctx context.Context, ps Params) (Response, error) {
+	ltp := ps.TablePaths.LeftTablePath
+	rtp := ps.TablePaths.RightTablePath
+	btp := ps.TablePaths.BaseTablePath
+	s3Creds := ps.S3Creds
+	dr, err := d.client.TableDiff(ctx, &DiffRequest{
+		Props: &DiffProps{
+			Repo: ps.Repo,
+			LeftTablePath: &TablePath{
+				Path: ltp.Path,
+				Ref:  ltp.Ref,
+			},
+			RightTablePath: &TablePath{
+				Path: rtp.Path,
+				Ref:  rtp.Ref,
+			},
+			BaseTablePath: &TablePath{
+				Path: btp.Path,
+				Ref:  btp.Ref,
+			},
+		},
+		GatewayConfig: &GatewayConfig{
+			Key:      s3Creds.Key,
+			Secret:   s3Creds.Secret,
+			Endpoint: s3Creds.Endpoint,
+		},
+	})
+	if err != nil {
+		errStatus, _ := status.FromError(err)
+		if codes.NotFound == errStatus.Code() {
+			return Response{}, ErrTableNotFound
+		}
+		return Response{}, err
+	}
+	return Response{
+		Diffs:      buildDiffEntries(dr),
+		ChangeType: ChangeType(dr.ChangeType),
+	}, nil
+}
+
+func buildDiffEntries(dr *DiffResponse) []DiffEntry {
```

Comment:
nit; should we decide between record/entry and go with one term?

Reply:
decided 😄